CIP-????/README.md

```diff
@@ -0,0 +1,168 @@
+Using the eUTxO model, it is currently possible to create spending scripts that validate based off of the transaction as a whole instead of any individual UTxO. Imagine if three UTxOs were spent from such a script's address within the same transaction. Which datum should be passed to the spending script? Currently, the ledger-script interface passes each UTxO's datum to the script individually and executes this "transaction level" spending script three times. Since the transaction context does not change between executions, these additional executions are completely redundant. There is currently no way to use a transaction level spending script so that these redundant executions do not happen. This limitation significantly handicaps Cardano's usage of the eUTxO model. While there is a potential workaround, the workaround has its own major drawbacks. You can read more about it in the related CPS [here](https://github.com/fallen-icarus/CIPs/tree/full-eUTxO-potential/CPS-%3F%3F%3F%3F#readme).
+
+Note: While Cardano simple scripts do not use a datum, they may also suffer from this limitation.
+
+### Problem 2: Wrong/Missing Datums Result in Permanently Locked Script UTxOs
+When a script's UTxO is missing a datum or has the wrong datum, the plutus spending script is unable to parse the datum attached to the UTxO and is therefore guaranteed to fail. This results in permanently locked UTxOs at the script's address. In order to help prevent accidental locking of script UTxOs, proxies are generally used. The idea is for users to send their funds to this proxy first. The proxy then attaches the required datum for the users and passes the funds onto the desired script address. However, it is very difficult to use such proxies in a decentralized way. In short, by being forced to use proxies, the Dapp loses some of its decentralization.
+
+Further, there are a number of use cases where a datum is not needed for the Dapp to function. In these situations, a dummy datum is still required just to prevent locking.
+
+As an addition point, regulators are not going to like any application where it is easy to accidentally lock (lose) users' funds permanentally. Regulators may force certain certifications for Dapp developers as a consequence. The motivation is exactly the same for why finanical advisors need to be licensed. Currently, this ease of accidental locking is the default on Cardano. This needs to be changed if we do not want regulators forcing developer certifications that are inherently centralizing.
+
+### Problem 3: No Universal Plutus Script
+Since plutus spending scripts take a datum while plutus minting and staking scripts do not, it is currently not possible to create a universal plutus script (one that can do all three). There are use cases where the Dapp would be more secure if the same script could be used for both spending and minting. An example is to only mint a token if a deposit is made to the relevant script address. 
```

Comment:
Consider the following example (this is an actual example I have encountered). I have a Dapp spending script that can be paired with any staking key/script. However, I would like to be able to tag all addresses that were made using the spending script with a dedicated token in order to tell them apart from other addresses. The requirements:
1. The token cannot be minted to an address that is not using the Dapp's spending script.
2. Once the token is at the Dapp's address, the token cannot be withdrawn. It can only be burned.

The solution to (1) is to hard code the spending script hash into the minting policy. Now the minting policy can check that the token is never minted to an address not using the spending script. The solution to (2) is to require the minting policy ID to be attached to the token's output in the datum. Now the spending script can prevent that token from leaving since it knows the policy ID.

The problem is in number 2. What if a user deposits a utxo with a datum containing a different policy ID? While this does not allow misuse of the Dapp's token, if the token with the supplied policy ID is present and not burnable by the user, that utxo is now locked forever since the spending script will not allow withdrawing that token. The spending script can only go off of what is in the datum.

This issue would not exist if the same script could both mint and spend since the script would automatically know the proper policy ID. Saying it would be "more secure" may not be the most appropriate description for this but IMO neither is saying it would be "easier to write". Perhaps saying it would be "easier to guarantee proper usage" would be better.